Scripts/Run-SS-On-Exp/scripts/growth_rate_stats.py

```diff
@@ -0,0 +1,295 @@
+
+    return x_co_vertex, y_co_vertex
+
+
+def find_endpoints(cells):
+    # fit ellipse to micro colony
+    # endpoints
+    vertex1_x = cells['x_end_point1'].values.tolist()
+    vertex1_y = cells['y_end_point1'].values.tolist()
+    vertex2_x = cells['x_end_point2'].values.tolist()
+    vertex2_y = cells['y_end_point2'].values.tolist()
+
+    bacteria_center_x = cells['x_center'].values.tolist()
+    bacteria_center_y = cells['y_center'].values.tolist()
+    bacteria_minor = cells['minor'].values.tolist()
+    bacteria_orientation = cells['orientation'].values.tolist()
+
+    # now I calculate co-vertexes of bacteria ellipse
+    co_vertex_x, co_vertex_y = find_co_vertex(bacteria_center_x, bacteria_center_y, bacteria_minor,
+                                              bacteria_orientation)
+
+    endpoints_x = vertex1_x + vertex2_x + co_vertex_x
+    endpoints_y = vertex1_y + vertex2_y + co_vertex_y
+    endpoints = pd.DataFrame(zip(endpoints_x, endpoints_y))
+
+    return endpoints
+
+
+def get_distances_to_colony_edge(cells, center, major, minor, theta):
+    """
+    Obtain distances to colony edge for a list of cells
+
+    @param  cells   cellStates dict
+    @param  center  center of the fitted ellipse (x, y)
+    @param  major   major axis length of the fitted ellipse
+    @param  minor   minor axis length of the fitted ellipse
+    @parap  theta   angle of rotation of the fitted ellipse
+    @return dist    nparray of shortest distances to colony border along each cell's major axis
+    """
+
+    # Initialize storage variables
+    n_cells = cells.shape[0]
+    dist = np.zeros(n_cells)
+
+    # Fit ellipse around colony
+    # center, major, minor, theta = welzl(all_centroids)
+
+    # For each cell, calculate min distance to colony border in direction of cell's major axis
+    for i, cell in cells.iterrows():
+        cell_centroid = [cells.iloc[i]['x_center'], cells.iloc[i]['y_center']]
+        function_inputs = (center, major, minor, theta, cell)
+        roots = solve_roots(function_inputs)
+        dist[i] = shortest_distance(roots[0], roots[1], cell_centroid)
+
+    return dist
+
+
+def calc_dist_vs_growth_rate(cells):
+    """
+    The main function that will be used by pyabc
+
+    Takes as input a cellStates dict and outputs the slope of the linear fit to
+    distance to colony border along cell's main axis vs. growth rate
+
+    @param  cells      dataframe         cells features value dataframe
+    @param  fig_export_path     optional: path to export a plot of growth rate vs. distance to colony edge
+    @return dist_vs_growth_rate slope of linear fit to distance vs. growth rate
+    """
+    # Create cells dataframe filtered for cellAge > 1
+    cells_age_filtered = filter_by_cell_age(cells, min_age=2)
+
+    if cells_age_filtered.shape[0] > 0:
+
+        # Get single-cell properties
+        growth_rates = cells_age_filtered['growth_rate'].values.tolist()
```

Comment:
This will not work with CellModeller data since it isn't in dataFrame format.

Reply:
In the comments below I talked about it completely :)